src/rcv/septentrio.c

```diff
@@ -99,824 +217,1580 @@ static uint8_t sig_tbl[SBF_MAXSIG+1][2]={ /* system, obs-code */
-        sprintf(raw->msgtype,"SBF %4d (%4d): %s",type,raw->len,tstr);
-    }
-    switch (type) {
-        case SBF_MEASEPOCH : return decode_measepoch (raw);
-        case SBF_MEASEXTRA : return decode_measextra (raw);
-        case SBF_GPSRAWCA  : return decode_gpsrawca  (raw);
-        case SBF_GLORAWCA  : return decode_glorawca  (raw);
-        case SBF_GALRAWFNAV: return decode_galrawfnav(raw);
-        case SBF_GALRAWINAV: return decode_galrawinav(raw);
-        case SBF_GEORAWL1  : return decode_georawl1  (raw);
-        case SBF_BDSRAW    : return decode_bdsraw    (raw);
-        case SBF_QZSRAWL1CA: return decode_qzsrawl1ca(raw);
-        case SBF_NAVICRAW  : return decode_navicraw  (raw);
-    }
-    trace(3,"sbf unsupported message: type=%d\n",type);
+static int decode_sbf(raw_t *raw) {
+  uint8_t *p = raw->buff;
+  uint32_t week, tow;
+  char tstr[32];
+  int type = U2(p + 4) & 0x1fff;
+  uint16_t crc;
+
+  crc = rtk_crc16(p + 4, raw->len - 4);
+  if (crc != U2(p + 2) || crc == 0) {
+    trace(2, "sbf crc error: type=%d len=%d\n", type, raw->len);
+    return -1;
+  }
+  if (raw->len < 14) {
+    trace(2, "sbf length error: type=%d len=%d\n", type, raw->len);
+    return -1;
+  }
+  tow = U4(p + 8);
+  week = U2(p + 12);
+  if (tow == 4294967295u || week == 65535u) {
+    trace(2, "sbf tow/week error: type=%d len=%d\n", type, raw->len);
+    return -1;
+  }
+  raw->time = gpst2time(week, tow * 0.001);
+
+  if (raw->outtype) {
+    time2str(raw->time, tstr, 2);
+    sprintf(raw->msgtype, "SBF %4d (%4d): %s", type, raw->len, tstr);
+  }
+  trace(3, "decode_sbf: type=%04x len=%d\n", type, raw->len);
+  switch (type) {
+  case SBF_MEASEPOCH:
+    return decode_measepoch(raw);
+  case SBF_MEASEXTRA:
+    return decode_measextra(raw);
+  case SBF_GPSRAWCA:
+    return decode_gpsrawca(raw);
+  case SBF_GLORAWCA:
+    return decode_glorawca(raw);
+  case SBF_GALRAWFNAV:
+    return decode_galrawfnav(raw);
+  case SBF_GALRAWINAV:
+    return decode_galrawinav(raw);
+  case SBF_GEORAWL1:
+    /*  case SBF_GEORAWL5: Needs testing*/
+    return decode_georawl1(raw);
+  case SBF_BDSRAW:
+    return decode_bdsraw(raw);
+  case SBF_QZSRAWL1CA:
+    return decode_qzsrawl1ca(raw);
+  case SBF_NAVICRAW:
+    return decode_navicraw(raw);
+
+  case SBF_GPSNAV:
+    return decode_gpsnav(raw);
+  case SBF_GPSION:
+    return decode_gpsion(raw);
+  case SBF_GPSUTC:
+    return decode_gpsutc(raw);
+  case SBF_GPSALM:
+    return decode_gpsalm(raw);
+  case SBF_GPSRAWL2C:
+  case SBF_GPSRAWL5:
+    return decode_rawnav(raw, SYS_GPS);
+
+  case SBF_GEONAV:
+    return decode_sbasnav(raw);
+  case SBF_GLONAV:
+    return decode_glonav(raw);
+  case SBF_GLOTIME:
+    return decode_gloutc(raw);
+  case SBF_GALNAV:
+    return decode_galnav(raw);
+  case SBF_GALION:
+    return decode_galion(raw);
+  case SBF_GALUTC:
+    return decode_galutc(raw);
+  case SBF_GALALM:
+    return decode_galalm(raw);
+
+    /* Ones we _know_ we wont handle*/
```

Comment:
To reduce noise in logs, made catches for all of the messages that exist in the file that we dont care about.